.github/workflows/go-test.yaml

```diff
@@ -26,6 +26,17 @@ jobs:
         id: go
       - name: Check out code
         uses: actions/checkout@v3
+      - name: Check uncommitted file changes
+        run: |
+          make generate manifests
+          clean=$(git status --porcelain)
+          if [[ -z "$clean" ]]; then
+              echo "Empty git status --porcelain: $clean"
+          else
+              echo "Uncommitted file changes detected: $clean"
+              git diff
+              exit 1
+          fi
```

Comment:
I like this, but maybe we should have it in a separate PR. I don't see it as part of the 'Unit Tests' workflow.

Reply:
No problem, should I take it out of this PR and open another one?

Reply:
Open another one is up to you of course. For now, if you can please take it out of this PR, I think it's cleaner. Thanks!

Reply:
I removed that commit from the PR history.